src/DIRAC/Resources/Computing/ComputingElement.py

```diff
@@ -91,20 +91,13 @@ def setToken(self, token, valid=0):
 
     def _prepareProxy(self):
         """Set the environment variable X509_USER_PROXY"""
-        if not self.proxy:
```

Comment:
Why do we completely remove this block?
I suspect that `self.proxy` is, or at least should be, always defined, is it correct?

Then may be we could return an error if it is not defined, such as:

```python
if not self.proxy:
    return S_ERROR("No proxy available")

# Continue with line95-line101
```

What do you think?

Reply:
If proxy is defined, self.proxy will not be None. This is simply a protection agains non-existing proxy which is a valid case

Reply:
> If proxy is defined, self.proxy will not be None
Ok fine.

But with your PR, if I execute `_prepareProxy()` without having a proxy, I will still get `S_OK()`.
It seems misleading.

Reply:
Why would you execute _prepareProxy without proxy ? This does not make sense in this case, that's why the protection. The outcome of _prepareProxy is irrelevant, it would mean a bug in the code that should be spotted while testing.